### PR TITLE
copr: support more ops

### PR DESCRIPTION
**Pull request by BusyJay**
Support minus, intdiv and mod ops.  Multiply will be supported in next pr, which will also impl Mul for Decimal.

@hhkbp2 @coocood PTAL

### CODE REVIEW

File: src/util/codec/datum.rs [hunk 1]
```diff
@@ -457,6 +457,125 @@ impl Datum {
...
+        let res = match (self, d) {
+            (Datum::I64(l), Datum::I64(r)) => l.checked_sub(r).into(),
+            (Datum::I64(l), Datum::U64(r)) => {
+                if l < 0 {
+                    Datum::Null
+                } else {
+                    (l as u64).checked_sub(r).into()
+                }
+            }
+            (Datum::U64(r), Datum::I64(l)) => {
```
Comment (coocood) [inline src/util/codec/datum.rs]: Why use `r`, `l` rather than `l`, `r`?

**Comment by coocood** [comment]
LGTM

**Comment by BusyJay** [comment]
@hhkbp2 PTAL

File: src/util/codec/datum.rs [hunk 2]
```diff
@@ -457,6 +457,125 @@ impl Datum {
...
+            (Datum::I64(l), Datum::I64(r)) => {
+                match l.checked_div(r) {
+                    None => Err(box_err!("{} intdiv {} overflow", l, r)),
+                    Some(res) => Ok(Datum::I64(res)),
+                }
+            }
+            (Datum::I64(l), Datum::U64(r)) => {
+                if l < 0 {
+                    if opp_neg!(l) >= r {
+                        Err(box_err!("{} intdiv {} overflow", l, r))
```
Comment (hhkbp2) [inline src/util/codec/datum.rs]: Precisely speaking, underflow.

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**